lib/nimble_options.ex

```diff
@@ -555,6 +567,18 @@ defmodule NimbleOptions do
     end
   end
 
+  defp validate_type(:map, key, value) do
+    if is_map(value) and keyword_list?(Map.to_list(value)) do
+      {:ok, value}
+    else
+      error_tuple(
+        key,
+        value,
+        "expected #{inspect(key)} to be a map, got: #{inspect(value)}"
```

Comment:
I think this needs a better error message.

Let's use `cond` so that we can check `is_map` + `keyword_list?` separately than just `is_map`.

```elixir
cond do
  is_map(value) and keyword_list?(Map.to_list(value)) → ...
  is_map(value) → ...
  true → ...
end
```

Reply:
I refactored this a little so we should get better error messages for this case, but its now covered in the fact that `:map` is and alias to `{:map, :atom, :any}` so we will get an error if the value is not a map, and a distinct error if the keys are not atoms

Reply:
Additionally I added logic to generate the specs in the docs for both of these as well